### ISSUE TITLE
Validation Bypass

### ISSUE THREAD

**Issue by Garrestocles** [manvel-khnkoyan/jpv#10]
Hello,

I'm a security researcher at Sonatype, and I discovered a potential vulnerability in this project. Do you have a preferred way for me to share the details privately, or do you want me to just show you what I've got on this GitHub issue?

**Comment by manvel-khnkoyan** [comment]
Please, could you share it here ?
Thanks

**Comment by Garrestocles** [comment]
I was looking at issue https://github.com/manvel-khnkoyan/jpv/issues/6, and it seems like we can still bypass the provided fix. In [the line where it's checking if the constructors match](https://github.com/manvel-khnkoyan/jpv/blob/8a3cb7793e07532866f1d64e56082eadf9881ae4/index.js#L274), it's still possible to just set the constructor of the data we're trying to get validated to mock whatever it's supposed to be.

Here's a Proof of Concept:
```
const jpv = require('jpv');

const someJson = {
    definitelyAnArray: {
      sneakyStuff: "Don't tell anyone, but I'm not actually an array.", 
      constructor: [].constructor
    }
};
const schema = {
  definitelyAnArray: []
};

// jpv.validate(someJson, schema) should return false, but, as of 2.2.1, returns true
console.log("Validation is getting bypassed: " + jpv.validate(someJson, schema)); 
```

**Comment by manvel-khnkoyan** [comment]
The issue was fixed #10
